tests/unit/libbpf_test_jit.cpp

```diff
@@ -245,1584 +147,284 @@ test_invalid_bpf_action(char log_buffer[])
-
-    // Get info on the new fd.
-    memset(&attr, 0, sizeof(attr));
-    attr.info.bpf_fd = fd3;
-    attr.info.info = (uintptr_t)&info;
-    attr.info.info_len = sizeof(info);
-    REQUIRE(bpf(BPF_OBJ_GET_INFO_BY_FD, &attr, sizeof(attr)) == 0);
-    REQUIRE(info.id == id1);
-
-    // Get info on the second link.
-    memset(&attr, 0, sizeof(attr));
-    info = {};
-    attr.info.bpf_fd = fd2;
-    attr.info.info = (uintptr_t)&info;
-    attr.info.info_len = sizeof(info);
-    REQUIRE(bpf(BPF_OBJ_GET_INFO_BY_FD, &attr, sizeof(attr)) == 0);
-    REQUIRE(info.type == BPF_LINK_TYPE_PLAIN);
-    REQUIRE(info.id == id2);
-    REQUIRE(info.prog_id != 0);
-
-    // And for completeness, try an invalid bpf() call.
-    REQUIRE(bpf(-1, &attr, sizeof(attr)) == -EINVAL);
-
-    // Unpin the link.
-    REQUIRE(ebpf_object_unpin("MyPath") == EBPF_SUCCESS);
-
-    Platform::_close(fd1);
-    Platform::_close(fd2);
-    Platform::_close(fd3);
-}
-
-TEST_CASE("enumerate link IDs with bpf", "[libbpf][bpf]")
-{
-    test_enumerate_link_IDs_with_bpf();   
-}
-
-void
-test_bpf_prog_attach()
-{
-    _test_helper_libbpf test_helper;
-    test_helper.initialize();
-
-    struct bpf_object* object = bpf_object__open("cgroup_sock_addr.o");
-    REQUIRE(object != nullptr);
-
-    struct bpf_program* program = bpf_object__find_program_by_name(object, "authorize_connect4");
-    REQUIRE(program != nullptr);
-
-    REQUIRE(bpf_object__load(object) == 0);
-
-    int program_fd = bpf_program__fd(program);
-    REQUIRE(program_fd > 0);
-
-    // Verify we can't attach the program to an attach type that doesn't work with this API.
-    REQUIRE(bpf_prog_attach(program_fd, 0, BPF_ATTACH_TYPE_SAMPLE, 0) == -ENOTSUP);
-
-    // Verify we can't use an illegal program fd.
-    REQUIRE(bpf_prog_attach(ebpf_fd_invalid, 0, BPF_CGROUP_INET4_CONNECT, 0) == -EBADF);
-
-    // TODO (issue #1028): Currently one can pass an invalid attachable fd and bpf_prog_attach
-    // will succeed because it's temporarily just treated as a compartment id. The following
-    // should instead return errors.
-    REQUIRE(bpf_prog_attach(program_fd, ebpf_fd_invalid, BPF_CGROUP_INET4_CONNECT, 0) == 0);
-    uint32_t link_id;
-    REQUIRE(bpf_link_get_next_id(0, &link_id) == 0);
-    fd_t link_fd = bpf_link_get_fd_by_id(link_id);
-    REQUIRE(link_fd >= 0);
-    REQUIRE(bpf_link_detach(link_fd) == 0);
-    REQUIRE(bpf_prog_attach(program_fd, program_fd, BPF_CGROUP_INET4_CONNECT, 0) == 0);
+    struct bpf_object* object;
+    int program_fd;
+#pragma warning(suppress : 4996) // deprecated
+    int result = bpf_prog_load_deprecated("test_sample_ebpf.o", BPF_PROG_TYPE_SAMPLE, &object, &program_fd);
+    REQUIRE(result == 0);
+    REQUIRE(object != nullptr);
+    REQUIRE(program_fd != ebpf_fd_invalid);
 
     bpf_object__close(object);
 }
 
-TEST_CASE("bpf_prog_attach", "[libbpf]")
-{
-   test_bpf_prog_attach();
-}
+TEST_CASE("libbpf load program", "[libbpf][deprecated]") { test_libbpf_load_program(); }
```

Comment:
We should probably have JIT or something in the test case names in this file, to make it clear they're not for native.